src/libtelio.udl

```diff
@@ -0,0 +1,572 @@
+    /// // Connects to VPN exit node, with specified allowed_ips.
+    /// telio_connect_to_exit_node_with_id(
+    ///     "5e0009e1-75cf-4406-b9ce-0cbb4ea50366",
+    ///     "QKyApX/ewza7QEbC03Yt8t2ghu6nV5/rve/ZJvsecXo=",
+    ///     "100.100.0.0/16;10.10.23.0/24",
+    ///     "1.2.3.4:5678"
+    /// );
+    ///
+    /// // Connect to exit peer via DERP
+    /// telio_connect_to_exit_node_with_id(
+    ///     "5e0009e1-75cf-4406-b9ce-0cbb4ea50366",
+    ///     "QKyApX/ewza7QEbC03Yt8t2ghu6nV5/rve/ZJvsecXo=",
+    ///     "0.0.0.0/0",
+    ///     NULL
+    /// );
+    /// ```
+    ///
+    [Throws=TelioError]
+    void connect_to_exit_node_with_id(string? identifier, PublicKey public_key, sequence<IpNetwork>? allowed_ips, SocketAddr? endpoint);
+    
+    /// Disconnects from specified exit node.
+    ///
+    /// # Parameters
+    /// - `public_key`: WireGuard public key for exit node.
+    ///
+    [Throws=TelioError]
+    void disconnect_from_exit_node([ByRef] PublicKey public_key);
+    
+    /// Disconnects from all exit nodes with no parameters required.
+    [Throws=TelioError]
+    void disconnect_from_exit_nodes();
+
+    /// Enables magic DNS if it was not enabled yet,
+    ///
+    /// Routing should be set by the user accordingly.
+    ///
+    /// # Parameters
+    /// - 'forward_servers': List of DNS servers to route the requests trough.
+    ///
+    /// # Examples
+    ///
+    /// ```c
+    /// // Enable magic dns with some forward servers
+    /// telio_enable_magic_dns("[\"1.1.1.1\", \"8.8.8.8\"]");
+    ///
+    /// // Enable magic dns with no forward server
+    /// telio_enable_magic_dns("[\"\"]");
+    /// ```
+    [Throws=TelioError]
+    void enable_magic_dns([ByRef] sequence<IpAddr> forward_servers);
+    
+    /// Disables magic DNS if it was enabled.
+    [Throws=TelioError]
+    void disable_magic_dns();
+
+    /// Enables meshnet if it is not enabled yet.
+    /// In case meshnet is enabled, this updates the peer map with the specified one.
+    ///
+    /// # Parameters
+    /// - `cfg`: Output of GET /v1/meshnet/machines/{machineIdentifier}/map
+    ///
+    [Throws=TelioError]
+    void set_meshnet(Config cfg);
+    
+    /// Disables the meshnet functionality by closing all the connections.
+    [Throws=TelioError]
+    void set_meshnet_off();
+    
+    
+    sequence<Node> get_status_map();
+    
+    /// Get last error's message length, including trailing null
+    string get_last_error();
+
+    /// For testing only.
+    [Throws=TelioError]
+    void generate_stack_panic();
+    
+    /// For testing only.
+    [Throws=TelioError]
+    void generate_thread_panic();
+};
+
+callback interface TelioEventCb {
+    void event(string payload);
+};
+
+callback interface TelioLoggerCb {
+    void log(TelioLogLevel log_level, string payload);
+};
+
+callback interface TelioProtectCb {
+    void protect(i32 payload);
+};
+
+/// Encompasses all of the possible features that can be enabled
+dictionary Features {
```

Comment:
I imagine that this can make going back and forth between libtelio versions slightly harder, but that is mostly a concern for the apps dev teams so I think they should also be involved here.